### DISCUSSION TITLE
The $record is updated for Select columns without saving

### DISCUSSION THREAD

**Discussion by zbadshah** [filamentphp/filament#5521] [Help]
Used the function    protected function handleRecordUpdate(Model $record, array $data): Model.

The Select column is used as 
                Forms\Components\Select::make('sales_id')
                    ->label('Sales User ID')
                    ->relationship('sales', 'name'),

The DB is getting updated when the above function gets called without using the $record->update($data) command.

Can somebody comment why is this happening?

**Comment by danharrin** [comment]
All relationship fields save their own data to the record. Same as repeaters and mutliselects and stuff

**Reply by zbadshah** [reply]
OK. Thanks. Then is there a way to get original data before modification in such cases?

**Reply by danharrin** [reply]
What do you want to do?

**Reply by zbadshah** [reply]
If a user changes the sales id then I need to take an action and save old id in the log.

**Reply by danharrin** [reply]
use a model observer?

**Reply by zbadshah** [reply]
Thanks a lot.
